using-timescaledb/continuous-aggregates.md

```diff
@@ -248,6 +248,124 @@ hypertable with a continuous aggregate.
+
+Usually, continuous aggregates are defined on a time column, but it is
+also possible to create your own custom scheme for handling
+aggregation for tables that are using an integer for the time
+column. This can be useful if you have tables that use other measures
+of time that can be represented as integer values, such as nanosecond
+epochs, minutes since founding date, or whatever is suitable for your
+application.
+
+As an example, suppose that you have a table with CPU and disk usage
+for some devices where time is measured in
+[microfortnights][fff-system]. Since we are using an integer-valued
+column as time, we need to provide the chunk time interval when
+creating the hypertable. In this case, we let each chunk consist of a
+millifortnight.
+
+```sql
+CREATE TABLE devices(
+  time BIGINT,        -- Timestamp in microfortnights
+  cpu_usage INTEGER,  -- Total CPU usage
+  disk_usage INTEGER, -- Total disk usage
+  PRIMARY KEY (time)
+);
+
+SELECT * FROM create_hypertable('devices', 'time',
+                                chunk_time_interval => 1000);
+```
+
+To define a continuous aggregate on a table that is using an integer
+dimension, it is necessary to have a function to get the current
+timestamp in whatever representation that you are using and set it for
+the hypertable using
+[`set_integer_now_func`][api-set-integer-now-func]. The function can
+be defined as a normal Postgres function, but need to be
+[`STABLE`][pg-func-stable], take no arguments, and should return an
+integer of the same type as the time column in the table. In our case,
+this should suffice:
+
+```sql
+CREATE OR REPLACE FUNCTION current_microfortnight() RETURNS BIGINT
+LANGUAGE SQL STABLE AS $$
+	SELECT CAST(1209600 * EXTRACT(EPOCH FROM CURRENT_TIMESTAMP) / 1000000 AS BIGINT)
+$$;
+
+SELECT set_integer_now_func('devices', 'current_microfortnight');
+```
+
+Once the replacement for current timestamp has been set up, you can
+define the continuous aggregate for the `devices` table.
+
+```sql
+CREATE VIEW devices_summary
+WITH (timescaledb.continuous,
+      timescaledb.refresh_interval = '1 minute',
+      timescaledb.refresh_lag = '-500') AS
+SELECT time_bucket('500', time) AS bucket,
+       avg(cpu_usage) AS avg_cpu,
+       avg(disk_usage) AS avg_disk
+     FROM devices
+    GROUP BY bucket;
+```
+
+You can now insert some rows to check if the aggregation works as
+expected. In this case, we are using the `tablefunc` extension to
+generate a Gaussian distribution of CPU usage and a cumulative growth
+of disk usage.
+
+```sql
+CREATE EXTENSION tablefunc;
+
+INSERT INTO devices(time, cpu_usage, disk_usage)
+SELECT time,
+       normal_rand(1,70,10) AS cpu_usage,
+	   normal_rand(1,2,1) * (row_number() over()) AS disk_usage
+  FROM generate_series(1,10000) AS time;
+```
+
+You can now check that the view contain the correct data.
+
+```sql
+postgres=# SELECT * FROM devices_summary ORDER BY bucket;
```

Comment:
Maybe trim the output, LIMIT to 10 rows or something.

Reply:
Done.